### PR TITLE
Improve nucliadb node test performance

### PR DESCRIPTION
**Pull request by jotare**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb_protos/python/nucliadb_protos/nodesidecar_pb2_grpc.pyi [hunk 1]
```diff
@@ -12,6 +12,7 @@ from nucliadb_protos.noderesources_pb2 import (
     IndexMetadata as IndexMetadata,
     IndexParagraph as IndexParagraph,
     IndexParagraphs as IndexParagraphs,
+    NodeMetadata as NodeMetadata,
```
Comment (lferran) [inline nucliadb_protos/python/nucliadb_protos/nodesidecar_pb2_grpc.pyi:15]: does it belong to this PR?

File: nucliadb_protos/python/nucliadb_protos/nodesidecar_pb2.pyi [hunk 2]
```diff
@@ -18,6 +18,7 @@ from nucliadb_protos.noderesources_pb2 import (
     IndexMetadata as IndexMetadata,
     IndexParagraph as IndexParagraph,
     IndexParagraphs as IndexParagraphs,
+    NodeMetadata as NodeMetadata,
```
Comment (lferran) [inline nucliadb_protos/python/nucliadb_protos/nodesidecar_pb2.pyi:21]: same: does this belong here?

File: nucliadb_node/nucliadb_node/tests/fixtures.py [hunk 3]
```diff
@@ -214,16 +229,17 @@ def data_path():
 
 
 @pytest.fixture(scope="function")
-def shadow_folder(data_path):
+def shadow_folder(data_path) -> Iterable[str]:
```
Comment (lferran) [inline nucliadb_node/nucliadb_node/tests/fixtures.py:232]: NIT: typing would be simpler if we simply return the string

File: nucliadb_node/nucliadb_node/pull.py [hunk 4]
```diff
@@ -70,12 +70,19 @@ def __init__(
...
         for subscription in self.subscriptions:
             try:
                 await subscription.drain()
             except nats.errors.ConnectionClosedError:
                 pass
         self.subscriptions = []
+
+        try:
+            await self.nc.close()
+        except (RuntimeError, AttributeError):  # pragma: no cover
```
Comment (lferran) [inline nucliadb_node/nucliadb_node/pull.py:83]: Can you comment why we could have `RuntimeError` and `AttributeError`

File: nucliadb_protos/python/nucliadb_protos/nodesidecar_pb2_grpc.pyi (hunk 1 above)
Comment (jotare) [reply nucliadb_protos/python/nucliadb_protos/nodesidecar_pb2_grpc.pyi:15]: No, it's missing from management of entities. I probably forgot to rebuild protos after one of the last rebases

File: nucliadb_protos/python/nucliadb_protos/nodesidecar_pb2.pyi (hunk 2 above)
Comment (jotare) [reply nucliadb_protos/python/nucliadb_protos/nodesidecar_pb2.pyi:21]: Same as above

**Review by lferran — APPROVED** [review]
LGTM! Thanks for the cleanup!
Just a few comments

File: nucliadb_node/nucliadb_node/tests/fixtures.py (hunk 3 above)
Comment (jotare) [reply nucliadb_node/nucliadb_node/tests/fixtures.py:232]: Yes, but as it's an async generator, mypy complains if you annotate only with `str`

Comment (lferran) [reply nucliadb_node/nucliadb_node/tests/fixtures.py:232]: What I meant is that you can change the `yield` to a return and then the annotation just a `str`

File: nucliadb_node/nucliadb_node/pull.py (hunk 4 above)
Comment (jotare) [reply nucliadb_node/nucliadb_node/pull.py:83]: They were handled on NATS utility. RuntimeError can occurr if the event loop is closed. I'm not sure about the AttributeError though. I'll try to know it and write down a comment

**Pull request merged**